doc/ug/getting_started_build_sw.md

```diff
@@ -95,6 +95,159 @@ $ ./meson_init -t /path/to/toolchain/file
 If you do not specify your own toolchain configuration file (using `./meson_init.sh -t`), and `meson_init.sh` cannot find the default configuration in your toolchain, the legacy `toolchain.txt` from the main OpenTitan repository will be used.
 If `TOOLCHAIN_PATH` is set, this will be used to update any paths within the legacy configuration.
 
+## Building Software with Bazel
+
+The Mask ROM and many other software targets are built with
+(Bazel)[https://bazel.build/]
+
+TLDR:
+after installing Bazel you can run
+
+```console
+bazel test //... --test_tag_filters=-cw310 --disk_cache=~/bazel_cache
+```
+
+### Installing Bazel
+
+If you haven't yet installed Bazel, add it to apt and instal it on ubuntu systems  with the following
+commands:
+
+```console
+sudo apt install apt-transport-https curl gnupg
+curl -fsSL https://bazel.build/bazel-release.pub.gpg | gpg --dearmor > bazel.gpg
+sudo mv bazel.gpg /etc/apt/trusted.gpg.d/
+echo "deb [arch=amd64] https://storage.googleapis.com/bazel-apt stable jdk1.8" |
+sudo tee /etc/apt/sources.list.d/bazel.list
+sudo apt update && sudo apt install bazel-4.2.0
+```
+
+or by following [instructions for your system](https://bazel.build/install)
+though it may not be supported by OpenTitan.
+
+### Building software with Bazel
+
+You can then build and run all the tests all the Bazel rules for OpenTitan within the workspace with the command
+
+```console
+bazel test //...
+```
+
+This is likely to include unhealthy tests and targets and slow tests and targets
+so you're likely to want to run more specific builds and tests, but this
```

Comment:
```suggestion
so you're likely to want to run more specific builds and tests, however this
```

Reply:
Sorry for the comma splice. Unwrapped sentences really shows off my run-on sentences.